### PR TITLE
Add FQDN support for device onboarding

### PR DESCRIPTION
**Pull request by susanhooks**
Added support to use FQDN to onboard devices in _validate_ip_addresses function, formatting. 
Updated job description. 
Addresses feature request #241

### CODE REVIEW

File: nautobot_device_onboarding/diffsync/adapters/sync_devices_adapters.py [hunk 1]
```diff
@@ -297,8 +331,16 @@ def load_device_types(self):
...
+        required_fields_from_device = [
+            "device_type",
+            "hostname",
+            "mgmt_interface",
+            "mask_length",
+            "serial",
+        ]
+        if (
+            platform
+        ):  # platform is only returned with device data if not provided on the job form/csv
```
Comment (jeffkala) [inline nautobot_device_onboarding/diffsync/adapters/sync_devices_adapters.py:343]: Did ruff autoformat do this?  This looks wonky.

File: nautobot_device_onboarding/diffsync/adapters/sync_devices_adapters.py [hunk 2]
```diff
@@ -311,7 +353,9 @@ def load_devices(self):
         for ip_address in self.device_data:
             if self.job.debug:
                 self.job.logger.debug(f"loading device data for {ip_address}")
-            platform = None  # If an excption is caught below, the platform must still be set.
+            platform = (
+                None  # If an excption is caught below, the platform must still be set.
+            )
```
Comment (jeffkala) [inline nautobot_device_onboarding/diffsync/adapters/sync_devices_adapters.py:358]: Did ruff autoformat do this?  This looks wonky.

**Review by jeffkala — COMMENTED** [review]
Looks good.  I'm a bit confused by some of the formatting changes.  Almost like ruff isn't using the correct profile or something?

**Review by scetron — CHANGES_REQUESTED** [review]
I think this just needs to have the formatting looked at as there's also a lot of diffs that are just a format change. Additionally the format that was changed looks incorrect.

**Review by scetron — APPROVED** [review]

**Pull request merged**